src/auto-evo/simulation/food_source/EnvironmentalFoodSource.cs

```diff
@@ -0,0 +1,47 @@
+﻿public class EnvironmentalFoodSource : IFoodSource
+{
+    private readonly Compound compound;
+    private readonly Compound glucose = SimulationParameters.Instance.GetCompound("glucose");
+
+    private BiomeConditions biomeConditions;
+    private float totalEnvironmentalEnergySource;
+
+    public EnvironmentalFoodSource(Patch patch, string compound, float foodCapacityMultiplier)
+    {
+        biomeConditions = patch.Biome;
+        this.compound = SimulationParameters.Instance.GetCompound(compound);
+        totalEnvironmentalEnergySource = patch.Biome.Compounds[this.compound].Dissolved * foodCapacityMultiplier;
+    }
+
+    public float FitnessScore(Species species)
+    {
+        var microbeSpecies = (MicrobeSpecies)species;
+
+        var energyCreationScore = 0.0f;
+        foreach (var organelle in microbeSpecies.Organelles)
+        {
+            foreach (var process in organelle.Definition.RunnableProcesses)
+            {
+                if (process.Process.Inputs.ContainsKey(compound))
+                {
+                    if (process.Process.Outputs.ContainsKey(glucose))
+                    {
+                        energyCreationScore += process.Process.Outputs[glucose]
+                            / process.Process.Inputs[compound] / Constants.AUTO_EVO_GLUCOSE_USE_SCORE_DIVISOR;
```

Comment:
Why is `AUTO_EVO_GLUCOSE_USE_SCORE_DIVISOR` a divisor and not a multiplier like `0.5f`? Floating point divides are much more expensive than multiplies, and it is very easy to convert constants to multiply operations.